fdbrpc/sim2.actor.cpp

```diff
@@ -945,6 +945,13 @@ class Sim2 final : public ISimulator, public INetworkConnections {
 	                        const std::vector<NetworkAddress>& addresses) override {
 		mockDNS.addMockTCPEndpoint(host, service, addresses);
 	}
+	void removeMockTCPEndpoint(const std::string& host, const std::string& service) override {
+		mockDNS.removeMockTCPEndpoint(host, service);
+	}
+	const std::map<std::string, std::vector<NetworkAddress>> getMockDNS() override { return mockDNS.getMockDNS(); }
+	void setMockDNS(std::map<std::string, std::vector<NetworkAddress>> otherMockDNS) override {
```

Comment:
Depending on how you plan to use it. This `setMockDNS` just looks too powerful. If you plan to reset the MockDNS, a `reset()` maybe a better interface. In your test, you can then have a `setMockDNS` function to call `reset` and then `addMockTCPEndpoint`

Reply:
I've refactored a little bit, as at the high level, we only need parseMockDNSFromString() and convertMockDNSToString() in network.